mangaki/mangaki/management/commands/compare.py

```diff
@@ -1,121 +1,85 @@
-import csv
-import os.path
-from collections import Counter
-
-import numpy as np
 from django.conf import settings
 from django.core.management.base import BaseCommand
-from sklearn.metrics import mean_squared_error
-from sklearn.model_selection import train_test_split
 
+from collections import defaultdict
+from sklearn.model_selection import ShuffleSplit
+import numpy as np
+import csv
+import os.path
+
+from mangaki.utils.data import Dataset
 from mangaki.utils.wals import MangakiWALS
 from mangaki.utils.als import MangakiALS
 from mangaki.utils.knn import MangakiKNN
 from mangaki.utils.svd import MangakiSVD
 from mangaki.utils.pca import MangakiPCA
 from mangaki.utils.zero import MangakiZero
-
 from mangaki.utils.values import rating_values
+import logging
 
-import matplotlib.pyplot as plt
 
-TEST_SIZE = 0.2  # 20% is kept for the test set
+NB_SPLIT = 5  # Divide the dataset into 5 buckets
+FILENAMES = {
+    'movies': 'ratings-ml.csv',
+    'mangas': 'ratings.csv'
+}
+CONVERT_FUNCTIONS = {
```

Comment:
I really feel conversion functions are the best approach so far.
The problem is, Movielens data are ratings between 0 and 5, and Mangaki data is categorical.
Maybe this should be an issue, so we can continue to talk about it.